File: Code_Caprice/链表/203移除链表元素.cpp


struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

ListNode *removeElements(ListNode *head, int val)
{
    ListNode *dummyhead = new ListNode();
    dummyhead->next = head;
    ListNode *cur = dummyhead;
    while(cur->next!=nullptr){
        if(cur->next->val == val){
            ListNode *tmp = cur->next;
            cur->next = cur->next->next;
            delete tmp;
        }else{
            cur = cur->next;
        }
    }
    head = dummyhead->next;
    delete dummyhead;
    return head;
}

int main(){
    
}

File: Code_Caprice/linkedList/203移除链表元素.py
from __future__ import annotations
from typing import Optional

class ListNode:
    def __init__(self, val: int = 0, next: ListNode = None) -> None:
        self.val: int = val
        self.next: ListNode = next


def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
    # 实现移除链表元素的逻辑
    dummyHead:ListNode = ListNode()
    dummyHead.next = head
    cur:ListNode = dummyHead
    while cur.next:
        if cur.next.val == val :
            cur.next = cur.next.next
        else:
            cur = cur.next
    return dummyHead.next


